server/services/constructionService.go

```diff
@@ -166,13 +173,10 @@ func (service *constructionService) ConstructionMetadata(
 	}, nil
 }
 
-func (service *constructionService) computeData(options *constructionOptions) []byte {
-	if service.extension.isNativeCurrencySymbol(options.CurrencySymbol) {
-		return options.Data
-	}
-
-	// TODO: Handle in a future PR
-	return make([]byte, 0)
+func (service *constructionService) computeDataForCustomCurrencyTransfer(tokenIdentifier string, amount string) []byte {
```

Comment:
When we'll support more types, we'll extract this logic to a separate component.